src/BinSkim.Rules/RulesExtensionMethods.cs

```diff
@@ -130,10 +130,11 @@ private static string CreateLibraryKey(ObjectModuleDetails objectModuleDetail)
 
         private static readonly Dictionary<CryptoError, string> s_cryptoErrorToDescriptionMap = BuildCryptoErrorDescriptions();
 
-        private static Dictionary<CryptoError, string> BuildCryptoErrorDescriptions()
+        public static Dictionary<CryptoError, string> BuildCryptoErrorDescriptions()
```

Comment:
>public [](http://example.com/codeflow?start=8&length=6)

`CryptoError` is public,
this is descriptions for `CryptoError`, I think it is ok to have it as public.

Reply:
I don't know how nitty we are about class visibility modifiers, but I think making it internal and using 'InternalsVisibleTo' to specify that the test assembly can access would be ideal.  If code signing is used, this requires sticking the public key in a class attribute.  If there's no code signing then it's trivial.

My take: if you haven't taken a crack at this yet, I'd encourage giving it a shot.  But not a hill to die on if you run into a hard time figuring out the signing stuff.

https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.internalsvisibletoattribute?view=net-7.0

Reply:
fixed